builtin/provisioners/chef/resource_provisioner.go

```diff
@@ -222,13 +227,19 @@ func (r *ResourceProvisioner) decodeConfig(c *terraform.ResourceConfig) (*Provis
 	}
 
 	if p.ValidationKeyPath != "" {
-		keyPath, err := homedir.Expand(p.ValidationKeyPath)
+		vkeyPath, err := homedir.Expand(p.ValidationKeyPath)
 		if err != nil {
 			return nil, fmt.Errorf("Error expanding the validation key path: %v", err)
 		}
-		p.ValidationKeyPath = keyPath
-	}
-
+		p.ValidationKeyPath = vkeyPath
+	}
+        if p.SecretKeyPath != "" {
+                skeyPath, err := homedir.Expand(p.SecretKeyPath)
```

Comment:
they are in the same logical path, but when defining them, your not just defining the path to where the files are, your defining the direct path to the file itself